src/main/scala/mesosphere/marathon/core/group/impl/GroupManagerImpl.scala

```diff
@@ -111,6 +113,8 @@ class GroupManagerImpl(
     change: (RootGroup) => Future[Either[T, RootGroup]],
     version: Timestamp, force: Boolean, toKill: Map[PathId, Seq[Instance]]): Future[Either[T, DeploymentPlan]] = try {
 
+    groupUpdateSizeMetric.increment()
```

Comment:
You could just ask the queue for the size instead of counting up and down ;) E.g. see here:
https://github.com/mesosphere/marathon/commit/2d217b46e19ad3f6ec8d80cd8023e558207e5235#diff-963e559f5c3864cf5e3971eaa415662dR119

Reply:
Yeah, I did that in previous version of this PR... but minmax counter does not have method for setting value and I think I can reliably do it by incrementing and decrementing